### PR TITLE
Default condition should be in sorted order together with other items

### LINKED ISSUE #403: Buildifier linter wants select keys sorted
When I run gazelle and then [buildifier lint](https://github.com/bazelbuild/buildtools/tree/master/buildifier#linter) it prints out warnings that look like this:
```bash
./vendor/golang.org/x/crypto/blake2b/BUILD.bazel:23: unsorted-dict-items: Dictionary items are out of their lexicographical order. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#unsorted-dict-items)
```
The generated target looks like this:
```bzl

go_library(
    name = "go_default_library",
    srcs = [
        "blake2b.go",
        "blake2bAVX2_amd64.go",
        "blake2bAVX2_amd64.s",
        "blake2b_amd64.go",
        "blake2b_amd64.s",
        "blake2b_generic.go",
        "blake2b_ref.go",
        "blake2x.go",
        "register.go",
    ],
    importmap = "stash.atlassian.com/micros/voyager/vendor/golang.org/x/crypto/blake2b",
    importpath = "golang.org/x/crypto/blake2b",
    visibility = ["//visibility:public"],
    deps = select({
        "@io_bazel_rules_go//go/platform:amd64": [
            "//vendor/golang.org/x/sys/cpu:go_default_library",
        ],
        "//conditions:default": [],
    }),
)
```
When I run buildifier fix, it corrects the order like this:
```bzl

go_library(
    name = "go_default_library",
    srcs = [
        "blake2b.go",
        "blake2bAVX2_amd64.go",
        "blake2bAVX2_amd64.s",
        "blake2b_amd64.go",
        "blake2b_amd64.s",
        "blake2b_generic.go",
        "blake2b_ref.go",
        "blake2x.go",
        "register.go",
    ],
    importmap = "stash.atlassian.com/micros/voyager/vendor/golang.org/x/crypto/blake2b",
    importpath = "golang.org/x/crypto/blake2b",
    visibility = ["//visibility:public"],
    deps = select({
        "//conditions:default": [],
        "@io_bazel_rules_go//go/platform:amd64": [
            "//vendor/golang.org/x/sys/cpu:go_default_library",
        ],
    }),
)
```
i.e. it sorts the keys. Would be great to make gazelle generate sorted keys there to make the linter happy.

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #403.

### CODE REVIEW

File: rule/merge.go [hunk 1]
```diff
@@ -408,18 +405,10 @@ func squashDict(x, y *bzl.DictExpr) (*bzl.DictExpr, error) {
...
 	for k := range cases {
-		if k == "//conditions:default" {
-			haveDefault = true
-			continue
-		}
 		keys = append(keys, k)
 	}
 	sort.Strings(keys)
-	if haveDefault {
-		keys = append(keys, "//conditions:default") // must be last
```
Comment (ash2k) [inline rule/merge.go]: This comments basically tells me this change is wrong. But it is not clear why it **must** be last.

Comment (ash2k) [reply rule/merge.go]: Documentation for `select` is here https://docs.bazel.build/versions/master/be/functions.html#select

Comment (ash2k) [reply rule/merge.go]: `bazel test //...` passes. So this "must" does not have a test 🤷‍♂️

Comment (jayconrod) [reply rule/merge.go]: "should" would be a more precise word. There's no technical reason `"//conditions:default"` needs to be at the bottom, but it's where it usually goes.

**Comment by jayconrod** [comment]
Thanks for sending this PR, but this was an issue in buildifier which was fixed a few hours ago. No change needed here.

**Pull request closed without merging**